jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala

```diff
@@ -288,6 +312,17 @@ object XGBoost extends Serializable {
       useExternalMemory, missing)
   }
 
+  /**
+   * This method trains an XGBoost model on a distributed collection of labeled training points.
+   * In each partition of the training RDD a booster is trained on the subset of data in that
+   * partition. The model is then updated globally using an all-reduce communication model across
+   * Spark tasks.
+   *
+   * A [[RabitTracker]] is created to coordinate communication between the all-reduce worker nodes
+   * and to verify that training completes successfully.
+   *
+   * Refer to `trainWithRDD` for parameter descriptions.
```

Comment:
same here, 

First, we do not need to explain what is Rabit tracker, Rabit is a third-party library relied by XGBoost (not XGBoost Spark). Just like in Spark, do you really need to explain what is parquet in the code?

Second, the comments here are describing what the method is doing. 1) comments should be used to explain why or very complex what; 2) here there is no why or complext what to explain; the core method is `buildDistributedBoosters` whose body is built with `RDD.zipPartitions`...we do not need to explain what zipPartitions is doing for such a project whose user should be familiar with Spark already